### DISCUSSION TITLE
execute confirm-password-reset The API prompts me：message: Invalid or expired token？

### DISCUSSION THREAD

**Discussion by bingoku** [pocketbase/pocketbase#3789] [Q&A] [answered]
I really didn't figure out what the error of my code was, I got the token after logging in with my email address and password, and I used this token to change the password, why did it prompt me that the token expired?  
```dart
Future<void> execPbSelect() async {
  final email = "test@example.com";
  final password = 'mArjiw-hynpe0-xawvow';
  final getInfo =
      await pb.collection('profile').authWithPassword(email, password);
  print(pb.authStore.isValid);
  await Future.delayed(Duration(seconds: 1));
  await pb.collection('profile').confirmPasswordReset(
        pb.authStore.token,
        'mArjiw-hynpe0-xawvow22',
        'mArjiw-hynpe0-xawvow22',
      );
}
```

```bash
❯ dart run
Building package executable...
Built cli:cli.
true
Unhandled exception:
ClientException: {url: https://pocketbase.mypve.testdev.pub:8849/api/collections/profile/confirm-password-reset, isAbort: false, statusCode: 400, response: {code: 400, message: Something went wrong while processing your request., data: {token: {code: validation_invalid_token, message: Invalid or expired token.}}}, originalError: null}
#0      PocketBase.send (package:pocketbase/src/client.dart:236:9)
<asynchronous suspension>
#1      execPbSelect (package:cli/cli.dart:76:3)
<asynchronous suspension>
#2      main (file:///Users/testdev/StudioProjects/cli/bin/cli.dart:7:3)
<asynchronous suspension>
```

**Comment by ganigeorgiev** [comment, marked as answer]
`pb.authStore.token` is an auth token and will not work.

You need a confirm password reset token which is sent to the user with the [`requestPasswordReset()`](https://pocketbase.io/docs/api-records/#request-password-reset) email.

Note that if you already have the old password, you can send a direct user update request like:

```js
await pb.collection('profile').update(pb.authStore.model?.id, body: {
    'oldPassword':     'yourOldPassword',
    'password':        'yourNewPassword',
    'passwordConfirm': 'yourNewPasswordConfirmation',
})
```

**Reply by bingoku** [reply]
As a novice, I will sort out the idea of using Pocketbase: the token obtained after logging in with the email address and password, is it just to verify whether the login request is successful or not? To customize and do other things, you need to decode the token to obtain the payload in it, and then expand the information in the payload, such as adding the email element, regenerating the token, and depositing it again into pb.authStore.save After that, it can be used for other logic, right?

**Reply by ganigeorgiev** [reply]
I'm sorry, I don't understand what do you mean. You don't have to decode anything manually or "regenerate" the token. For the confirm password reset endpoint just use the token that comes with the `requestPasswordReset()` email.
